### PR TITLE
[CARBONDATA-3889] Cleanup code for carbondata-common module

### PR DESCRIPTION
**Pull request by QiangCai**
### Why is this PR needed?
1. Field can be 'final'
2. Redundant 'if' statement
3. Empty method
4. Redundant 'throws' clause
5. Anonymous type can be replaced with lambda
6. Typo
7. Redundant String operation

 ### What changes were proposed in this PR?
1. change field to be 'final'
2. change redundant 'if' statement
3. remove Empty method
4. remove redundant 'throws' clause
5. change anonymous type  to lambda
6. fix typo
7. remove redundant String operation
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1565/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3302/

**Comment by QiangCai** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1567/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3304/

File: common/src/main/java/org/apache/carbondata/common/logging/impl/ExtendedRollingFileAppender.java [hunk 1]
```diff
@@ -203,19 +194,16 @@ public void rollOver() {
...
-            cleanupInProgress = true;
-            try {
-              cleanLogs(startName, folderPath, maxBackupIndex);
-            } catch (Throwable e) {
-              // ignore any error
-              LogLog.error("Cleaning logs failed", e);
-            } finally {
-              cleanupInProgress = false;
-            }
+      Runnable r = () -> {
```
Comment (kevinjmh) [inline common/src/main/java/org/apache/carbondata/common/logging/impl/ExtendedRollingFileAppender.java:197]: original one maybe more friendly to read

Comment (QiangCai) [reply common/src/main/java/org/apache/carbondata/common/logging/impl/ExtendedRollingFileAppender.java]: Reverted

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1671/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3413/

**Comment by kevinjmh** [comment]
LGTM

**Pull request closed without merging**